File: backend/java/src/main/java/de/iits/elocalculatorbackend/player/repository/PlayerRepository.java
package de.iits.elocalculatorbackend.player.repository;

import de.iits.elocalculatorbackend.player.model.entity.Player;
import org.springframework.data.domain.Limit;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface PlayerRepository extends JpaRepository<Player, String> {
    List<Player> findByOrderByEloDesc(Limit limit);
}

File: backend/kotlin/src/main/kotlin/de/iits/elo/player/persistence/PlayerRepository.kt
package de.iits.elo.player.persistence

import de.iits.elo.player.model.entity.Player
import org.springframework.data.domain.Limit
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PlayerRepository : JpaRepository<Player, String> {

    fun findByOrderByEloDesc(limit: Limit): List<Player>
}
